packages/lib/src/components/ThreeDS2/components/DelegatedAuthentication/DelegatedAuthentication.tsx

```diff
@@ -0,0 +1,331 @@
+import { h } from 'preact';
+import UIElement from '../../../UIElement';
+import {
+    DelegatedAuthenticationError,
+    DelegatedAuthenticationInputData,
+    DelegatedAuthenticationOutputData,
+    DelegatedAuthenticationProps
+} from './types';
+import { decodeBase64, encodeBase64 } from './utils';
+import { useEffect, useMemo, useState } from 'preact/hooks';
+import { createDelegatedAuthenticationResolveData, DelegatedAuthenticationResolveData, isInIframe } from '../utils';
+import Button from '../../../internal/Button';
+import { PaymentAmount } from '../../../../types';
+import { httpPost } from '../../../../core/Services/http';
+import { DelegatedAuthenticationResultResponse } from '../../types';
+
+class DelegatedAuthentication extends UIElement<DelegatedAuthenticationProps> {
+    public static type = 'delegatedAuthentication';
+
+    render() {
+        return <DelegatedAuthenticationImpl {...this.props} />;
+    }
+}
+
+const DelegatedAuthenticationImpl = ({
+    token,
+    paymentData,
+    onComplete,
+    useOriginalFlow,
+    clientKey,
+    elementRef,
+    loadingContext,
+    challengeWindowSize
+}: DelegatedAuthenticationProps) => {
+    const [loading, setLoading] = useState<boolean>(true);
+    const [authenticationData, setAuthenticationData] = useState<DelegatedAuthenticationInputData>(null);
+    const [hasCookieForCredential, setHasCookieForCredential] = useState<boolean>();
+
+    // todo: this should be a part of the token data
+    const useCookiesInSpc = true;
+
+    const shouldDoSpc = useMemo(() => {
+        return isInIframe();
+    }, []);
+
+    useEffect(() => {
+        if (token) {
+            try {
+                const data = parseDelegatedAuthenticationData(token);
+                if (!data) {
+                    complete({ errorCode: DelegatedAuthenticationError.NoInputDataPresent });
+                }
+                setAuthenticationData(data);
+            } catch (e) {
+                complete({ errorCode: DelegatedAuthenticationError.Unknown, errorDescription: e.message });
+            }
+        } else {
+            complete({ errorCode: DelegatedAuthenticationError.NoInputDataPresent });
+        }
+    }, []);
+
+    useEffect(() => {
+        if (!authenticationData) {
+            return;
+        }
+
+        if (shouldDoSpc && !useCookiesInSpc) {
+            // we are in an iframe, so we do spc, but we don't want to use cookies
+            setHasCookieForCredential(false);
+            return;
+        }
+
+        const cookieForCredential = hasCookieForSomeCredential(authenticationData);
+        if (!cookieForCredential) {
+            console.error('No cookie matching credential');
+            complete({ errorCode: DelegatedAuthenticationError.NoBrowserCookie });
+            return;
+        }
+        setHasCookieForCredential(cookieForCredential);
+    }, [authenticationData]);
+
+    useEffect(() => {
+        if (!authenticationData) {
```

Comment:
This useEffect depends on hasCookieForCrediential but doesn't call it anywhere? Should the if perhaps be (!authenticationData && hasCookieForCredential) otherwise every time hasCookieForCrediential gets updated this will run regardless of it's value.